dotnet/Liberator.Driver/RatDriverMethods.cs

```diff
@@ -364,6 +357,50 @@ public void ClickLinkAndWait(By locator, [Optional, DefaultParameterValue(true)]
             }
         }
 
+        /// <summary>
+        /// Clicks on a link and waits for a new page to be loaded that contains a specified URL or part URL
+        /// </summary>
+        /// <param name="element">The element on which to click</param>
+        /// <param name="url">The partial URL to be waited for</param>
+        /// <param name="waitForTarget">(Optional parameter) Whether to wait for the cliackability of the target element</param>
+        public void ClickLinkAndWaitForUrl(IWebElement element, string url, [Optional, DefaultParameterValue(true)] bool waitForTarget)
+        {
+            Element = element;
+            try
+            {
+                LastPage = Driver.FindElement(By.TagName("html"));
+                ClickLink(element, true);
+                WaitForUrlToContain(url);
+            }
+            catch (Exception ex)
+            {
+                if (_debugLevel == EnumConsoleDebugLevel.Human) { Console.WriteLine("Failure during attempt to click a link which opens a page."); }
+                HandleErrors(ex);
+            }
+        }
+
+        /// <summary>
+        /// Clicks on a link and waits for a new page to be loaded that contains a specified URL or part URL
+        /// </summary>
+        /// <param name="locator">The locator for the element on which to click</param>
+        /// <param name="url">The partial URL to be waited for</param>
+        /// <param name="wait">(Optional parameter) Whether to wait for the cliackability of the target element</param>
+        public void ClickLinkAndWaitForUrl(By locator, string url, [Optional, DefaultParameterValue(true)] bool wait)
+        {

```

Comment:
Same thing as above with the wait boolean

Reply:
Changed